File: src/main/java/com/radixpro/enigma/xchg/domain/CalculationSettings.java
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 */

package com.radixpro.enigma.xchg.domain;

import com.radixpro.enigma.domain.config.Configuration;
import com.radixpro.enigma.domain.config.ConfiguredCelObject;
import com.radixpro.enigma.references.*;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class CalculationSettings {

   private final List<CelestialObjects> celBodies;
   private final HouseSystems houseSystem;
   private final Ayanamshas ayanamsha;
   private final boolean sidereal;
   private final boolean topocentric;


   public CalculationSettings(@NotNull final List<CelestialObjects> celBodies, @NotNull final HouseSystems houseSystem, @NotNull final Ayanamshas ayanamsha,
                              final boolean sidereal, final boolean topocentric) {
      this.celBodies = celBodies;
      this.houseSystem = houseSystem;
      this.ayanamsha = ayanamsha;
      this.sidereal = sidereal;
      this.topocentric = topocentric;
   }

   public CalculationSettings(final Configuration configuration) {
      this.celBodies = constructCelObjects(configuration);
      this.houseSystem = configuration.getAstronConfiguration().getHouseSystem();
      this.ayanamsha = configuration.getAstronConfiguration().getAyanamsha();
      this.sidereal = configuration.getAstronConfiguration().getEclipticProjection() == EclipticProjections.SIDEREAL;
      this.topocentric = configuration.getAstronConfiguration().getObserverPosition() == ObserverPositions.TOPOCENTRIC;
   }

   public List<CelestialObjects> getCelBodies() {
      return celBodies;
   }

   public HouseSystems getHouseSystem() {
      return houseSystem;
   }

   public Ayanamshas getAyanamsha() {
      return ayanamsha;
   }

   public boolean isSidereal() {
      return sidereal;
   }

   public boolean isTopocentric() {
      return topocentric;
   }

   private List<CelestialObjects> constructCelObjects(final Configuration configuration) {
      List<CelestialObjects> celObjects = new ArrayList<>();
      List<ConfiguredCelObject> confCelObjects = configuration.getAstronConfiguration().getCelObjects();
      for (ConfiguredCelObject confCelObject : confCelObjects) {
         celObjects.add(confCelObject.getCelObject());
      }
      return celObjects;
   }

}

File: src/main/java/com/radixpro/enigma/xchg/domain/CalculationSettings.kt
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 */
package com.radixpro.enigma.xchg.domain

import com.radixpro.enigma.domain.config.Configuration
import com.radixpro.enigma.references.*
import java.util.*

class CalculationSettings {
    val celBodies: List<CelestialObjects>
    val houseSystem: HouseSystems
    val ayanamsha: Ayanamshas
    val isSidereal: Boolean
    val isTopocentric: Boolean

    constructor(celBodies: List<CelestialObjects>, houseSystem: HouseSystems, ayanamsha: Ayanamshas,
                sidereal: Boolean, topocentric: Boolean) {
        this.celBodies = celBodies
        this.houseSystem = houseSystem
        this.ayanamsha = ayanamsha
        isSidereal = sidereal
        isTopocentric = topocentric
    }

    constructor(configuration: Configuration) {
        celBodies = constructCelObjects(configuration)
        houseSystem = configuration.astronConfiguration.houseSystem
        ayanamsha = configuration.astronConfiguration.ayanamsha
        isSidereal = configuration.astronConfiguration.eclipticProjection === EclipticProjections.SIDEREAL
        isTopocentric = configuration.astronConfiguration.observerPosition === ObserverPositions.TOPOCENTRIC
    }

    private fun constructCelObjects(configuration: Configuration): List<CelestialObjects> {
        val celObjects: MutableList<CelestialObjects> = ArrayList()
        val confCelObjects = configuration.astronConfiguration.celObjects
        for ((celObject) in confCelObjects) {
            celObjects.add(celObject)
        }
        return celObjects
    }
}